src/select.jsx

```diff
@@ -165,7 +180,7 @@ var classBase = React.createClass({
         this.onChange();
 
         if (this.state.open) {
-          this.isFocusing = true;
+          this.setState({isFocusing: !this.state.isFocusing});
```

Comment:
I originally avoided `setState` here because it's async and was hitting race conditions. Does this work in all browsers?